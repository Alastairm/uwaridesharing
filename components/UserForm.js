import React, { Component, Proptypes} from 'react';
import { AsyncStorage, StyleSheet, Text, View } from 'react-native';
import Button from 'react-native-button';
// import store from 'react-native-simple-store';
import UserInput from './UserInput.js';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
    width: 200,
  },
  buttonContainer: {
    padding:10,
    height:45,
    overflow:'hidden',
    borderRadius:4,
    backgroundColor: 'steelblue'}
});

export default class UserForm extends Component {
  constructor(props) {
    super(props);
      this.state = {
        userData: {
          'name': "",
          'email':"",
          'phone':"",
        },
        data: "",
      }
    this.onChange = this.onChange.bind(this);
    this.onNext = this.onNext.bind(this);
    this.onSave = this.onSave.bind(this);
  }
  onChange(field, val) {
    userData = this.state.userData;
    userData[field] = val;
    this.setState({userData: userData});
  }
  onNext() {
    this.props.navigator.push({
      component: CheckForm
    });
  }
  onSave() {
    AsyncStorage.setItem('userData', JSON.stringify(this.state.userData));
  }
  render() {
    return (
      <View style={styles.container}>
        <UserInput onChange={this.onChange} field ="name"/>
        <UserInput onChange={this.onChange} field ="email"/>
        <UserInput onChange={this.onChange} field ="phone"/>
        <Button
          containerStyle={styles.buttonContainer}
          style={{fontSize: 20, color: 'white'}}
          onPress={this.onNext}>
          Next Page
        </Button>
        <Button
          containerStyle={styles.buttonContainer}
          style={{fontSize: 20, color: 'white'}}
          onPress={this.onSave}>
          Save
        </Button>
      </View>
    );
  }
}


class CheckForm extends Component {
  constructor(props) {
    super(props);
    this.state = {data: ""}
    this.onBackPress = this.onBackPress.bind(this);
    this.onRead = this.onRead.bind(this);
  }
  onBackPress() {
    this.props.navigator.pop();
  }
  onRead() {
    AsyncStorage.getItem('userData')
    .then( (value) => {
        this.setState({data:value});
      }).done();
  }
  render() {
    return(
      <View>
        <Button
        containerStyle={styles.buttonContainer}
        style={{fontSize: 20, color: 'white'}}
        onPress={this.onBackPress}>
          Back
        </Button>
        <Button
          containerStyle={styles.buttonContainer}
          style={{fontSize: 20, color: 'white'}}
          onPress={this.onRead}>
            Read Form Data
        </Button>
        <Text>
        {this.state.data}
        </Text>
      </View>
    );
  }
}
