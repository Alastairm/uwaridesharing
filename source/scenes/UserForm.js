import React, { Component } from 'react';
import { AsyncStorage, Text } from 'react-native';
import {
  Container,
  Content,
  Form,
  Item,
  Input,
  Label,
  Button,
  Icon,
} from 'native-base';
import Map from './Map.js';

export default class UserForm extends Component {
  constructor(props) {
    super(props);
    // ToDO: Check if user information is already saved
    this.state = {
      user: {
        name: false,
        email: false,
        phone: false,
      },
      valid: false,
      showErrors: false,
    };
    this.onNext = this.onNext.bind(this);
    this.onChange = this.onChange.bind(this);
    this.save = this.save.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onNext() {
    // Navigator should use context instead of props.
    // eslint-disable-next-line
    this.props.navigator.push({
      component: Map,
    });
  }
  onChange(field, valid, val) {
    const user = this.state.user;
    user[field] = (valid ? val : false); // Only save value if valid, otherwise  false
    this.setState({ user });
  }
  async onSubmit() {
    const hasName = this.state.user.name;
    const hasEmail = this.state.user.email;
    const hasPhone = this.state.user.phone;
    if (hasName && hasEmail && hasPhone) { // If all fields are valid, save & goto next scene
      this.save();
      this.onNext();
    } else { // If not all fields are valid display error messages for invalid fields
      this.setState({ showErrors: true });
    }
  }
  async save() {
    await AsyncStorage.setItem('user.saved', 'true');
    await AsyncStorage.setItem('user.name', this.state.user.name);
    await AsyncStorage.setItem('user.email', this.state.user.email);
    await AsyncStorage.setItem('user.phone', this.state.user.phone);
  }

  focusNextField(nextField) {
    this[nextField].focus();
  }
  render() {
    return (
      <Container>
        <Content>
          <Form>
            <Item floatingLabel>
              <Icon active name= 'person' />
              <Label>Name</Label>
              <Input />
            </Item>
            <Item floatingLabel>
              <Icon active name= 'mail' />
              <Label>Email</Label>
              <Input />
            </Item>
            <Item floatingLabel>
              <Icon active name= 'home' />
              <Label>Phone</Label>
              <Input />
            </Item>
            <Button rounded large info onPress={this.onNext} >
              <Text> submit </Text>
            </Button>
          </Form>
        </Content>
      </Container>
    );
  }

}
