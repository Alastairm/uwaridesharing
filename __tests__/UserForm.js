import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import UserForm from '../source/scenes/UserForm.js';


test('Matches snapshot', () => {
  const render = renderer.create(
    <UserForm />,
  ).toJSON();
  expect(render).toMatchSnapshot();
});
