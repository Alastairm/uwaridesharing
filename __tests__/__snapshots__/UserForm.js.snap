// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={Object {}}
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "paddingRight": 8,
                  "top": 6,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <View
            style={
              Object {
                "left": 22,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 8,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                Name
              </Text>
            </Text>
          </View>
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onChangeTextValue={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 60,
                "lineHeight": 24,
                "marginLeft": undefined,
                "paddingLeft": null,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            virtual={undefined}
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "paddingRight": 8,
                  "top": 6,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <View
            style={
              Object {
                "left": 22,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 8,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                Email
              </Text>
            </Text>
          </View>
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 60,
                "lineHeight": 24,
                "marginLeft": undefined,
                "paddingLeft": null,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            virtual={undefined}
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "paddingRight": 8,
                  "top": 6,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <View
            style={
              Object {
                "left": 22,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 8,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
                virtual={undefined}
              >
                Mobile
              </Text>
            </Text>
          </View>
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 60,
                "lineHeight": 24,
                "marginLeft": undefined,
                "paddingLeft": null,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            virtual={undefined}
          />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#62B1F6",
            "borderColor": "#007aff",
            "borderRadius": 57,
            "borderWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 26,
            "paddingRight": 26,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
           submit 
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginLeft": -15,
              "marginRight": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
